shoop/core/models/_contacts.py

```diff
@@ -234,12 +239,18 @@ class Meta:
         verbose_name_plural = _('persons')
 
     def save(self, *args, **kwargs):
+        if self.first_name or self.last_name:
+            self.name = (" ".join([self.first_name, self.last_name])).strip()
```

Comment:
Actually, setting the `person.name` shouldn't be conditional, but it should always be set on save, even if there is no first and last name:

```python
names = [self.first_name, self.last_name]
self.name = " ".join(x for x in names if x)
```